autogen/agentchat/conversable_agent.py

```diff
@@ -2579,6 +2579,9 @@ def _decorator(func: F) -> F:
             elif not hasattr(func, "_name"):
                 func._name = func.__name__
 
+            if (func._name in self._function_map) or (self.llm_config and "tools" in self.llm_config.keys() and func._name in [tool["function"]["name"] for tool in self.llm_config["tools"]]):
```

Comment:
1. Do the name checking inside `register_function`, which only checkes the `function_map`. `register_function` is only the agent that executes the function.
2. Do the name checking for `llm_config` inside `update_tool_signature` and `update_function_signature`, which is only for the agent that calls the function.